contracts/base/ModuleManager.sol

```diff
@@ -94,6 +94,7 @@ abstract contract ModuleManager is SelfAuthorized, Executor, IModuleManager {
         bytes memory data,
         Enum.Operation operation
     ) internal returns (address guard, bytes32 guardHash) {
+        onBeforeExecTransactionFromModule(to, value, data, operation);
         guard = getModuleGuard();
 
         // Only whitelisted modules are allowed.
```

Comment:
```suggestion
        require(msg.sender != SENTINEL_MODULES && modules[msg.sender] != address(0), "GS104");

        onBeforeExecTransactionFromModule(to, value, data, operation);

        guard = getModuleGuard();
```

Reply:
Proposed slightly different statement reordering. (Note that I couldn't propose the full change because of GitHub things).

Reply:
As discussed, if we move the code as suggested then `onBeforeExecTransactionFromModule` gets called after validation logic so, it won't be exactly a pre-hook.